lib/notifications/notification.ex

```diff
@@ -186,4 +232,41 @@ defmodule Notifications.Notification do
       notification_user_maps
     )
   end
+
+  defp convert_from_db_notification(db_notification) do
+    basic_fields = %{
+      id: db_notification.id,
+      created_at: db_notification.created_at,
+      state: db_notification.state
+    }
+
+    block_waiver = Map.get(db_notification, :block_waiver)
+    bridge_movement = Map.get(db_notification, :bridge_movement)
+
+    detail_fields =
+      cond do
+        block_waiver ->
+          Map.from_struct(block_waiver) |> Map.delete(:id)
+
+        bridge_movement ->
+          {reason, end_time} =
+            if bridge_movement.status == :raised do
+              {:chelsea_st_bridge_raised, bridge_movement.lowering_time}
+            else
+              {:chelsea_st_bridge_lowered,
+               db_notification.created_at + @notification_expiration_threshold}
```

Comment:
Shouldn't `db_notification.end_time` already be set at this point?

Reply:
There is no `db_notification.end_time` field, the `end_time` is on `Notification` structs rather than `DbNotification`s. We do have some duplicate code here but given how small and specialized it is I don't quite think it's worth extracting.

Reply:
Ahh, okay.